runtime/src/comm/ofi/comm-ofi.c

```diff
@@ -3935,6 +3939,7 @@ void chpl_comm_execute_on_nb(c_nodeid_t node, c_sublocid_t subloc,
              (int) node, (int) subloc, (int) fid, arg, argSize);
 
   CHK_TRUE(node != chpl_nodeID); // handled by the locale model
+  CHK_TRUE(initialized);
```

Comment:
Seems a little strange to only check that we're initialized for `chpl_comm_execute_on_nb`, any reason to only check there? I also have mild concerns about this being non-atomic. Can we end up in a situation where the initializing thread finishes, but since this is non-atomic other threads will have a stale result?

Using an atomic also makes me a little more nervous about the cost of doing a check on every comm operation, though with the c11 backend an atomic read at least on x86 shouldn't be any more expensive than a regular read.

Reply:
Michael objected to checking on every comm operation so I only put it on this one which is the first one invoked when broadcasting the global vars. Good point about atomicity, I was thinking that the initialization completed before the other threads started running but that's not true. I can go either way on this, I can remove it entirely or make it atomic whichever you prefer. The particular issue of regarding the number of locales will be caught by a different check, but I have run into several bugs caused by trying to use an uninitialized `ofi`.

Reply:
Yeah, I'm also not sure about putting it everywhere, but the choice of 1/3 execute_on seemed curious. I'd rather put it just in the broadcast code directly if that's what we wanted to check.

Reply:
I'd be surprised if adding one atomic read to every operation has any noticeable effect but I'd want to measure it. I suppose I could add a call to see if the comm layer is initialized and call that from `chpl_comm_broadcast_global_vars`. That might be the best solution.

Reply:
An atomic read on something that's only set once probably doesn't have too high a cost (at least on x86 with c11 atomics.) Just for a point of reference though enabling comm diags (a contended atomic add for each comm event) can lead to like a 50% slowdown for comm heavy applications. So atomics in general can be expensive, but this particular use case should be ok but it's not a change I'd want to make without some more investigation.

Reply:
Removed the initialization check. Doing it properly would open a huge can of worms that isn't worth opening for this issue. The error will be caught when the number of locales specified on the command-line is compared against that determined by the OOB.